# This file defines parameters specific to the Maelstrom ROV
#
# When looking at the ROV from above, the thruster placement is:
#  ___    ___
# |   |__|   |
# |F1      F2|    N
# |    F5    |    ^
# |          | W<-|->E
# |    F6    |    v
# |F4      F3|    S
# |__________|
#   REAR END
#
# Using the compass shown:
# F1 is positive NE
# F2 is positive NW
# F3 is positive SW
# F4 is positive SE
# F5 is positive down
# F6 is positive down

num_control_dof: 5
num_thrusters: 6
thrust_configuration:
 [[ 0.7071,  0.7071, -0.7071, -0.7071,     0.0,    0.0], # Surge
  [ 0.7071, -0.7071, -0.7071,  0.7071,     0.0,    0.0], # Sway
  [    0.0,     0.0,     0.0,     0.0,     1.0,    1.0], # Heave
  [-0.0467, -0.0467,  0.0467,  0.0467, -0.1600, 0.1600], # Pitch
  [ 0.2143, -0.2143,  0.2143, -0.2143,     0.0,    0.0]] # Yaw

max_force_x: 69.85
max_force_y: 69.85
max_force_z: 49.39
max_torque_y: 5 # Only an estimate
max_torque_z: 7 # Only an estimate

scaling_force_x: 1
scaling_force_y: 1
scaling_force_z: 1
scaling_torque_y: 1
scaling_torque_z: 1

controller_frequency: 10
